modules/ldk/ldk_lib/src/lib.rs

```diff
@@ -551,6 +558,137 @@ pub unsafe extern "C" fn ldk_parse_p2p_lightning_message(
     }
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn ldk_decode_onion(data: *const u8, len: usize) -> *mut c_char {
+    struct TestEcdhSigner {
+        node_secret: SecretKey,
+    }
+    impl NodeSigner for TestEcdhSigner {
+        fn ecdh(
+            &self,
+            _recipient: Recipient,
+            other_key: &PublicKey,
+            tweak: Option<&Scalar>,
+        ) -> Result<SharedSecret, ()> {
+            let mut node_secret = self.node_secret.clone();
+            if let Some(tweak) = tweak {
+                node_secret = self.node_secret.mul_tweak(tweak).map_err(|_| ())?;
+            }
+            Ok(SharedSecret::new(other_key, &node_secret))
+        }
+        fn get_expanded_key(&self) -> ExpandedKey {
+            unreachable!()
+        }
+        fn get_node_id(&self, _recipient: Recipient) -> Result<PublicKey, ()> {
+            unreachable!()
+        }
+        fn sign_invoice(
+            &self,
+            _invoice: &RawBolt11Invoice,
+            _recipient: Recipient,
+        ) -> Result<RecoverableSignature, ()> {
+            unreachable!()
+        }
+        fn get_peer_storage_key(&self) -> PeerStorageKey {
+            unreachable!()
+        }
+        fn get_receive_auth_key(&self) -> ReceiveAuthKey {
+            unreachable!()
+        }
+        fn sign_bolt12_invoice(
+            &self,
+            _invoice: &UnsignedBolt12Invoice,
+        ) -> Result<schnorr::Signature, ()> {
+            unreachable!()
+        }
+        fn sign_gossip_message(&self, _msg: UnsignedGossipMessage) -> Result<Signature, ()> {
+            unreachable!()
+        }
+        fn sign_message(&self, _: &[u8]) -> Result<String, ()> {
+            unreachable!()
+        }
+    }
+
+    let data = std::slice::from_raw_parts(data, len);
+
+    if data.len() < 32 {
+        return str_to_c_string("");
+    }
+    let private_key = match SecretKey::from_slice(&data[0..32]) {
+        Ok(private_key) => private_key,
+        Err(_) => return str_to_c_string(""),
+    };
```

Comment:
Nit: it's simpler to use let-else for these types of assignments

```suggestion
    let Ok(private_key) = SecretKey::from_slice(&data[0..32]) else {
        return str_to_c_string("");
    };
```